pkg/config/parameters.go

```diff
@@ -64,6 +64,12 @@ func (g *GatewayParameters) Validate() error {
 		}
 		errorString = strings.TrimSpace(fmt.Sprintf("%s namespace required", errorString))
 	}
+	if len(g.ControllerName) == 0 {
+		if len(errorString) > 0 {
+			errorString += ","
+		}
+		errorString = strings.TrimSpace(fmt.Sprintf("%s controllerName required", errorString))
+	}
```

Comment:
Isn't this a little more complicated? If you're configuring Gateway API, isn't only ControllerName required, and the other two are optional now? And shouldn't we be warning people that configuring `name` and `namespace` is deprecated and will be removed?

Reply:
This PR introduces the usage of ControllerName. We couldn't have deprecated out the name/namespace since now it's actually being used. The next follow-up PR will remove the name/namespace and use the controllername to look up the Gateway. Those bits are spewed throughout the code.

Reply:
https://github.com/projectcontour/contour/issues/3862

Reply:
Ah, thanks, I was confused about the timing, nice work.